src/main/scala/org/broadinstitute/dsde/workbench/sam/schema/JndiSchemaDAO.scala

```diff
@@ -49,15 +49,98 @@ object JndiSchemaDAO {
   }
 }
 
-class JndiSchemaDAO(protected val directoryConfig: DirectoryConfig)(implicit executionContext: ExecutionContext) extends JndiSupport with DirectorySubjectNameSupport {
+class JndiSchemaDAO(protected val directoryConfig: DirectoryConfig, val schemaLockConfig: SchemaLockConfig)(implicit executionContext: ExecutionContext) extends JndiSupport with DirectorySubjectNameSupport with LazyLogging {
+
+  def initSchemaLock(): Future[Unit] = {
+    for {
+      _ <- createSchemaChangeLockSchema()
+      _ <- createOrgUnit(schemaLockOu)
+      _ <- insertSchemaLock()
+    } yield ()
+  }
+
+  def cleanupSchemaLock(): Future[Unit] = {
+    for {
+      _ <- removeSchemaLock()
+      _ <- removeSchemaChangeLockSchema()
```

Comment:
do we need to remove it? That will have its own concurrency issues right: sam101 has the lock, sam102 is waiting, sam101 finishes and removes the schema then sam102 wakes up but can't insert the lock because the schema is gone.

Reply:
my initial thought was to remove the schema in case it changed, but that's a bit premature so i'll get rid of that line